src/lib/components/composites/settings/project-settings/general/ProjectNameSettings.svelte

```diff
@@ -0,0 +1,124 @@
+    import { generateFieldMask } from "protobuf-fieldmask";
+    import type { ApiError } from "$lib/model/general";
+    import ErrorAlert from "$lib/components/composites/utils/ErrorAlert.svelte";
+    import { invalidate } from "$app/navigation";
+
+    interface Props {
+        projectId: string;
+        loadingProject: Promise<Project>;
+    }
+
+    const { projectId, loadingProject }: Props = $props();
+
+    let projectName: string = $state("");
+
+    let projectNameInput: Input;
+
+    let updateProjectError: ApiError | undefined = $state(undefined);
+
+    let loading = $state(true);
+
+    /**
+     * Loading the project name on mount.
+     */
+    onMount(async () => {
+        loadingProject
+            .then((project) => {
+                projectName = project.name;
+                loading = false;
+            })
+            .catch((error) => {
+                updateProjectError = {
+                    errorTitle: "Something went wrong while loading the project name.",
+                };
+                console.error(`Couldn't load the project name: ${error}`);
+            });
+    });
+
+    /**
+     * Changes the name of the project, if a non-blank project name is provided.
+     *
+     * Therefore, send a request to the backend to update the project with the
+     * provided name.
+     */
+    async function handleSubmit(event: Event) {
+        event.preventDefault();
+
+        const projectNameValid = projectNameInput.validate();
+        if (!projectNameValid) return;
+
+        const projectData: Partial<Project> = {
+            id: projectId,
+            name: projectNameInput.getValue(),
+        };
+        const fieldMask = generateFieldMask(projectData);
+
+        backendService
+            .updateProject({
+                project: Project.create(projectData),
+                mask: {
+                    paths: fieldMask,
+                },
+            })
+            .response.then(() => {
+                invalidate("data:getProjectById");
+            })
+            .catch((error) => {
+                updateProjectError = {
+                    errorTitle: "Something went wrong while updating the project name.",
+                };
+                console.error(`Couldn't update project: ${error}`);
+            });
+    }
+</script>
+
+<!--
+@component
+Component for changing the project name settings.
+
+Usage:
+```svelte
+    <ProjectNameSettings {loadingProject} {projectId}/>
+```
+-->
+<SettingsSection sectionTitle="General">
+    <form
+        id="project-update"
+        class="flex w-full max-w-xl flex-col items-center gap-2.5 md:h-fit md:max-w-xl md:flex-row md:items-start"
+        onsubmit={handleSubmit}
+    >
+        <Input
+            bind:this={projectNameInput}
+            class="h-full w-full"
+            disabled={loading}
+            inputId="project-name-input"
+            label="Project Name"
+            placeholder="Project Name"
```

Comment:
```suggestion
            placeholder={loading ? "Loading" : "New Project Name ..."}
```
What do you think of this change?

Reply:
This is a good suggestion. Is implemented

Reply:
IMO the dots at the end are not necessary. What are your thoughts?

Reply:
You are right. @mowi12 approved it.